jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PushImageStep.java

```diff
@@ -94,10 +96,48 @@
     }
   }
 
+  static ImmutableList<PushImageStep> makeListPushManifestList(
+      BuildContext buildContext,
+      ProgressEventDispatcher.Factory progressEventDispatcherFactory,
+      RegistryClient registryClient,
+      ManifestTemplate manifestList,
+      boolean manifestAlreadyExists)
+      throws IOException {
+    Set<String> tags = buildContext.getAllTargetImageTags();
+
+    EventHandlers eventHandlers = buildContext.getEventHandlers();
+    try (TimerEventDispatcher ignored =
+            new TimerEventDispatcher(eventHandlers, "Preparing manifest list pushers");
+        ProgressEventDispatcher progressEventDispatcher =
+            progressEventDispatcherFactory.create("launching manifest list pushers", tags.size())) {
+
+      if (JibSystemProperties.skipExistingImages() && manifestAlreadyExists) {
+        eventHandlers.dispatch(
+            LogEvent.info("Skipping pushing manifest list; manifest already exists."));
+        return ImmutableList.of();
+      }
+      DescriptorDigest manifestDigest = Digests.computeJsonDigest(manifestList);
+      return manifestList instanceof V22ManifestListTemplate
```

Comment:
We should have consistency in the way we determine single or multi-image building. And I think early short-circuiting makes sense. So instead of this `instanceof` check, I'd go with

```java
  static ImmutableList<PushImageStep> makeList(...) throws IOException {
    boolean singlePlatform = buildContext.getContainerConfiguration().getPlatforms().size() == 1;
    if (singlePlatform) {
      return ImmutableList.of(); // single image; no need to push a manifest list
    }
    Set<String> tags = buildContext.getAllTargetImageTags();
```
```